main.cpp

```diff
@@ -1,39 +1,172 @@
+#include "nwpwin.h"
+#include "nwpdlg.h"
 #include "main.h"
 #include "rc.h"
 
-int number_dialog::idd() const {
-	return IDD_NUMBER; 
+#include <iostream>
+#include <string>
+#include <sstream>
+
+tstring to_tstring(int value) 
+{
+    std::wstringstream wss;
+    wss << value;
+    return wss.str();
 }
-bool number_dialog::on_init_dialog(){
-	return true;
+
+int number_dialog::idd() const
+{
+    return IDD_NUMBER;
 }
-bool number_dialog::on_ok(){
-	return true;
+
+bool number_dialog::on_init_dialog()
+{
+    SetDlgItemInt(*this, IDC_EDIT1, number, FALSE);
+    return true;
 }
- 
-void main_window::on_paint(HDC hdc){
+
+bool number_dialog::on_ok()
+{
+    BOOL success;
+    int input = GetDlgItemInt(*this, IDC_EDIT1, &success, FALSE);
+    if (!success || input < 1 || input > 15) {
+        MessageBox(*this, _T("Broj izmedu 1 i 15"), _T("Error"), MB_ICONERROR);
+        return false;
+    }
+    number = input;
+    return true;
+}
+
+main_window::main_window()
+{
+    _tcscpy(font.lfFaceName, _T("Arial"));
+    font.lfWeight = FW_MEDIUM;
+
+    //Def font
+    HDC hdc = GetDC(nullptr);
+    font.lfHeight = -MulDiv(11, GetDeviceCaps(hdc, LOGPIXELSY), 96);
+    ReleaseDC(nullptr, hdc);
 }
 
-void main_window::on_command(int id){
-	switch(id){
-		case ID_FONT: 
-			break;
-		case ID_NUMBER: 
-			break;
-		case ID_EXIT: 
-			::DestroyWindow(*this); 
-			break;
-	}
+void main_window::on_paint(HDC hdc)
+{
+    RECT rect;
+    GetClientRect(*this, &rect);
+
+    //Postavljanje velicine brojeva
+    LOGFONT dynamic_font = font;
+    if (!custom_font_size) 
+    {
+        int min_dimension = min(rect.right, rect.bottom);
+        int font_size = min_dimension / (table_size + 1) / 2;
+
+        font_size = max(6, min(font_size, 50));
+        dynamic_font.lfHeight = -MulDiv(font_size, GetDeviceCaps(hdc, LOGPIXELSY), 96);

```

Comment:
Ideja o dinamičkoj veličini fonta je dobra, ali ovdje miješate stvari - `min_dimension` su pikseli, `font_size` su pointi.
Pravi način bi bio da "izmjerite" koliko je visok text za neki font (mijenjate point size) i odaberete željenu visinu - možete recimo koristiti [DrawText sa DT_CALCRECT](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-drawtext).